trace/opentel/opentelemetry_tracer.go

```diff
@@ -0,0 +1,92 @@
+package trace
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/graph-gophers/graphql-go/errors"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/graph-gophers/graphql-go/introspection"
+	gogqltrace "github.com/graph-gophers/graphql-go/trace"
```

Comment:
I'd rather reave this package as `trace` and name the other `oteltrace`. Most devs might be used with `trace` being part of OTEL, however, in the context of this library I would prefer if we didn't use `gogqltrace`. Also we use the `oteltrace` in only one place and the other one in four places.